server/src/Domain/Entities/BaseEntity.cs

```diff
@@ -1,22 +1,16 @@
-﻿using StudyZen.Domain.ValueObjects;
+﻿using StudyZen.Domain.Interfaces;
+using StudyZen.Domain.ValueObjects;
 
 namespace StudyZen.Domain.Entities;
 
-public abstract class BaseEntity
+public abstract class BaseEntity : IAuditable
 {
     public int Id { get; set; }
-    public UserActionStamp CreatedBy { get; init; }
-    public UserActionStamp UpdatedBy { get; set; }
+    public UserActionStamp CreatedBy { get; set; } = null!;
```

Comment:
Why do we use set instead of init here?

Reply:
To keep the logic out of domain layer and make it lightweight